### PR TITLE
LL-6534 Translations bugs

### PR DESCRIPTION
**Pull request by juan-cortes**
Unified pr to fix translation bugs reported, please keep your commits atomic or create sub prs to this branch if the task is more involved. If QA checks any of these and it's not clear enough what to test, please comment for clarification.

- [**LL-6575**](https://ledgerhq.atlassian.net/browse/LL-6575) Rename FR-CD to avoid line wrap
    - [x] Dev
    - [x] QA
- [**LL-6577**](https://ledgerhq.atlassian.net/browse/LL-6577) Support plural for updatable applications
    - [x] Dev
    - [x] QA
- [**LL-6684**](https://ledgerhq.atlassian.net/browse/LL-6684) LLD - Wording - Onboarding
  - [x] Dev
  - [x] QA
- [**LL-6686**](https://ledgerhq.atlassian.net/browse/LL-6686) LLD - Wording - Settings page
  - [x] Dev
  - [x] QA
- [**LL-6687**](https://ledgerhq.atlassian.net/browse/LL-6687) LLD - Wording - Compound
  - [x] Dev
  - [x] QA
- [**LL-6688**](https://ledgerhq.atlassian.net/browse/LL-6688) LLD - Wording - Buy Flow
  - [x] Dev
  - [x] QA
- [**LL-6685**](https://ledgerhq.atlassian.net/browse/LL-6685) LLD - Wording - Catalog
  - [x] Dev
  - [x] QA
- [LL-6573](https://ledgerhq.atlassian.net/browse/LL-6573) LLD - dates should be translatable
  - [x] Dev
  - [x] QA
- [LL-6583](https://ledgerhq.atlassian.net/browse/LL-6583) Full node landing text wrap
  - [x] Dev
  - [x] QA
- [**LL-6579**](https://ledgerhq.atlassian.net/browse/LL-6579) Localize asset distribution numbers
  - [x] Dev
  - [ ] QA

### Type

Bug Fix

### Context

https://ledgerhq.atlassian.net/browse/LL-6534

### Parts of the app affected / Test plan

Once we have them all or a bunch of them we can ship a build to the translation team for validation.

### CODE REVIEW

File: static/i18n/en/app.json [hunk 1]
```diff
@@ -1301,7 +1301,8 @@
       },
       "updatable": {
         "title": "Updates available",
-        "progressTitle": "{{number}} App updates",
+        "progressTitle": "{{number}} App update",
+        "progressTitle_plural": "{{number}} App updates",
```
Comment (ggilchrist-ledger) [inline static/i18n/en/app.json:1305]: Just to confirm that this isn't actually implemented anywhere in the codebase yet. Is that expected?

**Comment by ggilchrist-ledger** [comment]
The ordering in the lists is a bit funky. Is this deliberate? If not I will raise an associated bug

Screenshot 2021-08-09 at 12 13 09

**Comment by ggilchrist-ledger** [comment]
We've not made the region text box bigger, but I will assume that's deliberate since it would not match the other boxes on that page @juan-cortes

**Comment by juan-cortes** [comment]
> We've not made the region text box bigger, but I will assume that's deliberate since it would not match the other boxes on that page @juan-cortes

I guess we could do it, but it's more involved than just the rename that was described in the task, there's no element that's wrapping to a second line anymore.

**Comment by juan-cortes** [comment]
> The ordering in the lists is a bit funky. Is this deliberate? If not I will raise an associated bug
> 
> Screenshot 2021-08-09 at 12 13 09

I've made it sorted by default, it should be available on the latest commit

File: static/i18n/en/app.json (hunk 1 above)
Comment (juan-cortes) [reply static/i18n/en/app.json:1305]: This is the text shown on the Manager when you have updatable applications on the connected device, yo should see a box listing the applications that could be updated. If there's only one, the text should read "1 App update" and if there are more, "X App updates", same from french which was the cause of the original bug

Comment (ggilchrist-ledger) [reply static/i18n/en/app.json:1305]: Unfortuantely I'm not seeing that. It's still saying plural for one update 
Screenshot 2021-08-09 at 15 49 47

Comment (juan-cortes) [reply static/i18n/en/app.json:1305]: You are right, but as @gre said we shouldn't test it in french but rather in English.
I'll change this one but for the rest I'll rely on crowdin

Comment (juan-cortes) [reply static/i18n/en/app.json:1316]: You are right, I was missing part of the translations, I updated the pr

**Comment by jgiamporcaro-ledger** [comment]
Added new commit

--- 

- [LL-6685](https://ledgerhq.atlassian.net/browse/LL-6685) LLD - Wording - Catalog
  - [x] Dev
  - [ ] QA

**Comment by ggilchrist-ledger** [comment]
@jgiamporcaro-ledger  The first ‘previous’ is correctly translated but the ‘switch device’ isn’t. Also where is the previous in the first page?

**Comment by jgiamporcaro-ledger** [comment]
@ggilchrist-ledger Tested again today and everything is fine from my side 🤔 

- The first "previous" is displayed dynamically when the onboarding is relaunched. Tbh with you, I didn't succeed to show it using the regular way so to make my test I just locally update the code a bit to force the displaying state and ensure that the translation is managed. From my side, it is.
- Regarding the "switch device" label, I have tested it and it is correctly managed by our translation system. 

I don't know, maybe you didn't have the last version of the branch when you tested it yesterday, could you test again, please? If you continue to have an issue with this commit, we can schedule a quick call, maybe I'll be able to help.

File: src/renderer/components/Platform/AppDetails.js [hunk 2]
```diff
@@ -87,7 +84,10 @@ type Props = {
 
 const AppDetails = ({ manifest }: Props) => {
   const { t } = useTranslation();
-  const language = useSelector(languageSelector);
+  const description = t(
+    `manager.apps.content.${manifest.id}.shortDescription`,
```
Comment (ggilchrist-ledger) [inline src/renderer/components/Platform/AppDetails.js:88]: Small thing, most of these use quotes `"` not back ticks ` ` `

Comment (gre) [reply src/renderer/components/Platform/AppDetails.js:88]: backtick is necessary for the templating of ${manifest.id} here, so seems legit.

**Comment by gre** [comment]
@juan-cortes plz merge back develop in. sorry for test issue.

**Comment by jgiamporcaro-ledger** [comment]
Added new commit @juan-cortes

[LL-6573](https://ledgerhq.atlassian.net/browse/LL-6573) LLD - dates should be translatable
- [x] Dev
- [ ] QA

File: static/i18n/en/app.json [hunk 3]
```diff
@@ -1339,6 +1351,44 @@
       }
     },
     "apps": {
+      "content": {
+        "paraswap": {
+          "shortDescription": "Swap your crypto with ParaSwap that aggregates and provides the best quotes decentralised exchanges.",
```
Comment (ggilchrist-ledger) [inline static/i18n/en/app.json:1356]: Small grammatical thing: I think this should be 'Swap your crypto with ParaSwap that aggregates and provides the best quotes *on* decentralised exchanges.' I assume this was copied from somewhere so I can ask someone in product to confirm @juan-cortes

Comment (jgiamporcaro-ledger) [reply static/i18n/en/app.json:1356]: Yeah, I confirm that's the version we currently have in production.

File: src/renderer/components/Platform/AppDetails.js (hunk 2 above)
Comment (jgiamporcaro-ledger) [reply src/renderer/components/Platform/AppDetails.js:88]: Backticks are used in Javascript to inject variables to a string, that's why I use them here

File: static/i18n/en/app.json (hunk 3 above)
Comment (ggilchrist-ledger) [reply static/i18n/en/app.json:1356]: Ok I will raise in the ticket with a product owner

File: src/renderer/components/Platform/AppDetails.js (hunk 2 above)
Comment (ggilchrist-ledger) [reply src/renderer/components/Platform/AppDetails.js:88]: https://media.giphy.com/media/jUwpNzg9IcyrK/giphy.gif

Comment (jgiamporcaro-ledger) [reply src/renderer/components/Platform/AppDetails.js:88]: Sorry, I suppose that I forgot to refresh the page before sending my message, didn't see that gre already responsed.

**Review by achabrolin-ledger — APPROVED** [review]

**Comment by ggilchrist-ledger** [comment]
@juan-cortes @jgiamporcaro-ledger Testing finished for now. Let me know if there is more to test

File: src/renderer/components/FormattedDate.js [hunk 4]
```diff
@@ -7,6 +7,7 @@ type Props = {
   format?: string,
 };
 
+/* TODO: Replace this function with the useDateTimeFormat hook */
```
Comment (valpinkman) [inline src/renderer/components/FormattedDate.js:10]: the hook exists in the PR, is this still necessary ?

**Review by valpinkman — CHANGES_REQUESTED** [review]

**Review by valpinkman — APPROVED** [review]
Seen with @juan-cortes, the TODO is legit but not ready yet

File: src/renderer/components/FormattedDate.js (hunk 4 above)
Comment (jgiamporcaro-ledger) [reply src/renderer/components/FormattedDate.js:10]: The plan is to replace this function (FormattedDate) with the new hook introduced by this PR to move from the `moment` library to the native `Intl` Javascript API.

**Review by valpinkman — CHANGES_REQUESTED** [review]
CI not passing

File: src/renderer/components/FormattedDate.js (hunk 4 above)
Comment (valpinkman) [reply src/renderer/components/FormattedDate.js:10]: great

File: static/i18n/en/app.json (hunk 3 above)
Comment (jgiamporcaro-ledger) [reply static/i18n/en/app.json]: I included this fix to the PR.

**Comment by gre** [comment]
conflict with develop.

**Comment by jgiamporcaro-ledger** [comment]
I solved conflicts by rebasing the branch. ⚠️ ⚠️

**Pull request merged**